tests/integration/test_design.py

```diff
@@ -2738,3 +2738,49 @@ def test_surface_body_creation(modeler: Modeler):
     assert len(design.bodies) == 6
     assert not body.is_surface
     assert body.faces[0].area.m == pytest.approx(39.4784176044 * 2)
+
+
+def test_cached_bodies(modeler: Modeler):
+    """Test verifying that bodies are cached correctly.
+
+    Whenever a new body is created, modified etc. we should make sure that the cache is updated.
+    """
+    design = modeler.create_design("ModelingDemo")
+
+    # Define a sketch
+    origin = Point3D([0, 0, 10])
+    plane = Plane(origin, direction_x=[1, 0, 0], direction_y=[0, 1, 0])
+
+    # Create a sketch
+    sketch_box = Sketch(plane)
+    sketch_box.box(Point2D([20, 20]), 30 * UNITS.m, 30 * UNITS.m)
+
+    sketch_cylinder = Sketch(plane)
+    sketch_cylinder.circle(Point2D([20, 20]), 5 * UNITS.m)
+
+    design.extrude_sketch(name="BoxBody", sketch=sketch_box, distance=Distance(30, unit=UNITS.m))
+    design.extrude_sketch(
+        name="CylinderBody",
+        sketch=sketch_cylinder,
+        distance=Distance(60, unit=UNITS.m),
+    )
+
+    my_bodies = design.bodies
+    my_bodies_2 = design.bodies
+
+    # We should make sure that the object memory addresses are the same
+    for body1, body2 in zip(my_bodies, my_bodies_2):
+        assert body1 is body2  # We are comparing the memory addresses
```

Comment:
I like this way to test it. I don't actually see any other way to do it.

Reply:
Correct... Testing this has to be done at "object" comparison level. They need to be exactly the same.